patches/systemd/0002-Add-support-for-systemd-boot-for-Managarm.patch

```diff
@@ -0,0 +1,651 @@
+ src/core/main.c                          |  3 +++
+ src/core/manager.c                       |  1 +
+ src/core/mount.c                         |  9 ++++++++
+ src/journal/journald-native.c            |  2 ++
+ src/journal/journald-server.c            | 12 ++++++++---
+ src/journal/journald-syslog.c            |  2 ++
+ src/journal/journald.conf                |  6 +++---
+ src/libsystemd/sd-journal/journal-send.c |  1 +
+ src/shared/async.c                       |  4 ++++
+ src/shared/mount-setup.c                 | 10 +++++++++
+ src/shared/mount-util.c                  |  7 +++++++
+ src/udev/udev.conf                       |  2 +-
+ units/meson.build                        |  4 ++++
+ units/runsvr-usbhid.service              | 26 ++++++++++++++++++++++++
+ units/systemd-journald.service.in        | 10 ++++-----
+ units/systemd-udevd.service.in           | 26 ++++++++++++------------
+ 23 files changed, 124 insertions(+), 34 deletions(-)
+ create mode 100644 units/runsvr-usbhid.service
+
+diff --git a/src/basic/capability-util.c b/src/basic/capability-util.c
+index e9b41fe..ed8472c 100644
+--- a/src/basic/capability-util.c
++++ b/src/basic/capability-util.c
+@@ -333,6 +333,7 @@ int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities) {
+         if (prctl(PR_SET_KEEPCAPS, 0) < 0)
+                 return log_error_errno(errno, "Failed to disable keep capabilities flag: %m");
+ 
++#ifndef __managarm__
+         /* Drop all caps from the bounding set (as well as the inheritable/permitted/effective sets), except
+          * the ones we want to keep */
+         r = capability_bounding_set_drop(keep_capabilities, true);
+@@ -365,6 +366,7 @@ int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities) {
+                 if (cap_set_proc(d) < 0)
+                         return log_error_errno(errno, "Failed to increase capabilities: %m");
+         }
++#endif
+ 
+         return 0;
+ }
+diff --git a/src/basic/process-util.c b/src/basic/process-util.c
+index 6a1e496..c9f30bb 100644
+--- a/src/basic/process-util.c
++++ b/src/basic/process-util.c
+@@ -1621,6 +1621,8 @@ int safe_fork_full(
+                 }
+         }
+ 
++// TODO: HACK!
++#ifndef __managarm__
+         if (FLAGS_SET(flags, FORK_NEW_MOUNTNS | FORK_MOUNTNS_SLAVE)) {
+                 /* Optionally, make sure we never propagate mounts to the host. */
+                 if (mount(NULL, "/", NULL, MS_SLAVE | MS_REC, NULL) < 0) {
+@@ -1628,6 +1630,7 @@ int safe_fork_full(
+                         _exit(EXIT_FAILURE);
+                 }
+         }
++#endif
+ 
+         if (FLAGS_SET(flags, FORK_PRIVATE_TMP)) {
+                 assert(FLAGS_SET(flags, FORK_NEW_MOUNTNS));
+@@ -1973,8 +1976,8 @@ _noreturn_ void freeze(void) {
+         }
+ 
+         /* waitid() failed with an unexpected error, things are really borked. Freeze now! */
+-        for (;;)
+-                pause();
++        for (;;);
++                // pause();
```

Comment:
Why is this patched out? This is on an error path anyway, isn't it?

Reply:
A PR for this is posted, but with that implemented I hang during boot. Probably an issue of the implementation, but we can fix that and remove this piece of the patch later.